sdk/src/types/block/basic.rs

```diff
@@ -219,12 +213,12 @@ impl Packable for BasicBlock {
             return Err(Error::InvalidBlockKind(kind)).map_err(UnpackError::Packable);
         }
 
-        let data = BasicBlockData::unpack::<_, VERIFY>(unpacker, visitor)?;
+        let data = BasicBlockData::unpack::<_, VERIFY>(unpacker, protocol_params)?;
 
         let Signature::Ed25519(signature) = Signature::unpack::<_, VERIFY>(unpacker, &())?;
 
         let block = Self {
-            protocol_version,
+            protocol_params: protocol_params.clone(),
```

Comment:
doesnt look like this clone should be needed

Reply:
Protocol params provided to the fn are by reference!